app/server/api-generator/src/main/java/io/syndesis/server/api/generator/openapi/OpenApiConnectorGenerator.java

```diff
@@ -171,10 +173,10 @@ private Connector basicConnector(final ConnectorTemplate connectorTemplate, fina
             Optional<ConfigurationProperty> maybeProperty;
             switch (info.getApiVersion()) {
```

Comment:
Kinda the same thing here as above. Would reduce the number of object instantiations as well. I don't believe that the `DataShapeGenerator`s or `OpenApiPropertyGenerator` hold any state so they're prime for caching in a field.

Reply:
Agreed, cached those in a field